sdk/metric/internal/aggregate/sum.go

```diff
@@ -32,257 +33,182 @@ func newValueMap[N int64 | float64]() *valueMap[N] {
 	return &valueMap[N]{values: make(map[attribute.Set]N)}
 }
 
-func (s *valueMap[N]) Aggregate(value N, attr attribute.Set) {
+func (s *valueMap[N]) measure(_ context.Context, value N, attr attribute.Set) {
 	s.Lock()
 	s.values[attr] += value
 	s.Unlock()
 }
 
-// newDeltaSum returns an Aggregator that summarizes a set of measurements as
-// their arithmetic sum. Each sum is scoped by attributes and the aggregation
-// cycle the measurements were made in.
-//
-// The monotonic value is used to communicate the produced Aggregation is
-// monotonic or not. The returned Aggregator does not make any guarantees this
-// value is accurate. It is up to the caller to ensure it.
-//
-// Each aggregation cycle is treated independently. When the returned
-// Aggregator's Aggregation method is called it will reset all sums to zero.
-func newDeltaSum[N int64 | float64](monotonic bool) aggregator[N] {
-	return &deltaSum[N]{
+// newSum returns an aggregator that summarizes a set of measurements as their
+// arithmetic sum. Each sum is scoped by attributes and the aggregation cycle
+// the measurements were made in.
+func newSum[N int64 | float64](monotonic bool) *sum[N] {
+	return &sum[N]{
 		valueMap:  newValueMap[N](),
 		monotonic: monotonic,
 		start:     now(),
 	}
 }
 
-// deltaSum summarizes a set of measurements made in a single aggregation
-// cycle as their arithmetic sum.
-type deltaSum[N int64 | float64] struct {
+// sum summarizes a set of measurements made as their arithmetic sum.
+type sum[N int64 | float64] struct {
 	*valueMap[N]
 
 	monotonic bool
 	start     time.Time
 }
 
-func (s *deltaSum[N]) Aggregation() metricdata.Aggregation {
+func (s *sum[N]) delta(dest *metricdata.Aggregation) int {
+	t := now()
+
+	sData, _ := (*dest).(metricdata.Sum[N])
```

Comment:
I find the intention of this line not obvious. Initially one could think that it should be `sData := (*dest).(metricdata.Sum[N])` (which is of course not correct). 

How about changing to

```go
	sData, ok := (*dest).(metricdata.Sum[N])
	if !ok {
		// Initialize sData.
		sData.Temporality = metricdata.DeltaTemporality
		sData.IsMonotonic = s.monotonic
	}
```

It is a little more lines of code, but I think it would make the intention clearer. I also think the performance would be similar.

Reply:
Your are not guaranteed that when `ok == true` the temporality and monotonicity are correct.

Reply:
I see. Just consider adding some comment to say that "zero-value" of `sData` is a valid scenario... 😉

Reply:
Yeah, I can add them back :+1:. They seemed redundant and repetitive to me, but I have been looking at this for many days now :smile:.

Reply:
Thanks a lot 👍